src/RSocketServer.cpp

```diff
@@ -64,10 +71,18 @@ void RSocketServer::start(OnSetupConnection onSetupConnection) {
       ->start([ this, onSetupConnection = std::move(onSetupConnection) ](
           std::unique_ptr<DuplexConnection> connection,
           folly::EventBase & eventBase) {
+        auto* acceptor = setupResumeAcceptors_.get();
+
+        if (isShutdown_) {
+          // connection is getting out of scope and terminated
+          //
+          // if we created a new acceptor its ok, we aren't queuing anything new
```

Comment:
Why not do the `if (isShutdown_) { ... }` check before `auto acceptor = setupResumeAcceptors_.get();` then?